ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -859,6 +883,54 @@ pub mod tests {
         );
     }
 
+    #[test]
+    fn test_special_columns_empty() {
+        let ledger_path = get_tmp_ledger_path_auto_delete!();
+        let blockstore = Blockstore::open(ledger_path.path()).unwrap();
+
+        // Nothing has been inserted yet
+        assert!(blockstore.special_columns_empty().unwrap());
+
+        let num_entries = 1;
+        let max_slot = 9;
+        for slot in 0..=max_slot {
+            let entries = make_slot_entries_with_transactions(num_entries);
```

Comment:
This whole block below of generating entries / shreds / inserting shreds / writing transaction status is duplicated across several functions in both `blockstore_purge.rs` and `blockstore.rs`.

However, adding it here in order to avoid any potential merge conflicts with https://github.com/solana-labs/solana/pull/33419, we can deduplicate stuff later